modules/core/src/weaver/TestName.scala

```diff
@@ -0,0 +1,20 @@
+package weaver
+
+/**
+ * An identifier to a test in a suite.
+ *
+ * The implicit conversion from String is used as a mean
+ * for IDEs to detect the location of individual tests.
+ *
+ * The IDE is then able to request running a test at a specific
+ * location, using a CLI call. The framework will be able to lookup
+ * the test using the location (which should match with what the IDE
+ * knows)
+ */
+case class TestName(name: String, location: SourceLocation)
```

Comment:
@keynmol This is the main change in the PR. This allows to capture the location of tests, so that they can be looked-up by line number (which Intellij discovers by looking for the implicit conversions from "String" to "TestName").